content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,566 @@
+                .secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+
+        // Add the above connector to the WebClient instance
+        return WebClient.builder()
+                .clientConnector(new ReactorClientHttpConnector(httpClient))
+                .build();
+    }
+}
+```
+
+We can break the above configuration into three parts:
+
+* *SslContext*: We can define an `SslContext` using *SslContextBuilder* class and defining various Trust Strategies to trust the incoming certificate request. This will set the common SslContext to validate the secure HTTP requests.
+* *HttpClient*: Next we can define an *HttpClient* and overload various configurations or settings such as connection timeout, log level settings, follow redirection of HTTP requests, adding security context, etc.
+* *WebClient*: Finally, we can define the builder for WebClient instance and overload with the above HttpClient instance.
+
+Now, let’s look into a basic GET call to retrieve some data from a server or an API.
+
+We can *Autowire* the WebClient instance that we have defined above and make use of it in our service class.
+
+```java
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class WebClientUtil {
+
+    private final WebClient webClient;
+
+    public WebClient.ResponseSpec getFakeUsers() {
+        return webClient
+                .get()
+                .uri("https://randomuser.me/api/")
+                .retrieve();
+    }
+}
+```
+
+So we called the WebClient instance, made a GET request and retrieved a ResponseSpec instance as a repsonse which can later be converted to Mono or Flux.
+
+Now let’s quickly take a look into one POST call. WebClient allows us to define headers, content types, body, etc. We will just create a simple POST call but we will take a look into functional methods to deal with the backup methods or logic in case of errors or need retries.
+
+```java
+    public Mono<User> postUser(User user) {
+        return webClient
+                .post()
+                .uri("http://localhost:9000/api/users")
+                .header("Authorization", "Basic MY_PASSWORD")
+                .accept(MediaType.APPLICATION_JSON)
+                .body(Mono.just(user), User.class)
+                .retrieve()
+                .bodyToMono(User.class)
+                .log()
+                .retryWhen(Retry.backoff(10, Duration.ofSeconds(2)))
+                .onErrorReturn(new User())
+                .doOnError(throwable -> log.error("Result error out for POST user", throwable))
+                .doFinally(signalType -> log.info("Result Completed for POST User: {}", signalType));
+    }
+```
+
+If we receive the response in the first attempt, then it will simply return the response back. But due to some reason, if the API faces some issue, then it will try to re-attempt for a while after which it can return some indicated data or logs to denote that it has failed due to some reason.
+
+## Alternative server used in WebFlux
```

Comment:
Please update your section headers to use title case (www.titlecase.com).

Reply:
Updated the title.